jubatus/core/nearest_neighbor/euclid_lsh.cpp

```diff
@@ -105,6 +109,10 @@ void euclid_lsh::neighbor_row(
     const std::string& query_id,
     vector<pair<string, float> >& ids,
     uint64_t ret_num) const {
+  util::concurrent::scoped_rlock lk(get_const_table()->get_mutex());
+
+  /* table lock acquired; all subsequent table operations must be nolock */
+
   const pair<bool, uint64_t> maybe_index =
       get_const_table()->exact_match(query_id);
```

Comment:
I think this should be `exact_match_nolock`.
Otherwise it will deadlock (cf. #200)